src/v/coproc/ntp_context.h

```diff
@@ -26,8 +26,27 @@
 #include <absl/container/node_hash_map.h>
 
 #include <chrono>
+#include <exception>
+
 using namespace std::chrono_literals;
+
 namespace coproc {
+
+class script_failed_exception final : public std::exception {
```

Comment:
i think coproc should use `std::error_code` and error_category, etc instead of exceptions for the cases i could see. 

use `result<T>` types. plays naturally w/ the RPC layer too.

Reply:
@senior7515 not sure if you saw but, having this type of failure be an exception means that the pacemaker can handle it even though it percolates from the script_context.